src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -159,6 +169,9 @@ public void close() throws ConnectException {
       }
     } catch (IOException e) {
       throw new ConnectException("Error closing " + logFile, e);
+    } finally {
+      writer = null;
```

Comment:
I can't point to the actual lines, but nullifying reader and writer above, is redundant now that it's done within the `finally` block.